applications/ChimeraApplication/custom_processes/rotate_region_process.h

```diff
@@ -0,0 +1,245 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Authors:        Aditya Ghantasala, https://github.com/adityaghantasala
+// 					        Navaneeth K Narayanan
+//
+
+#ifndef ROTATE_REGION_PROCESS_H
+#define ROTATE_REGION_PROCESS_H
+
+// System includes
+#include <cmath>
+#include <iostream>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "containers/model.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/variables.h"
+#include "processes/process.h"
+#include "utilities/math_utils.h"
+#include "utilities/quaternion.h"
+
+// Application includes
+#include "chimera_application_variables.h"
+
+namespace Kratos {
+
+/**
+ * @class RotateRegionProcess
+ * @ingroup KratosChimeraApplication
+ * @brief This is the to apply rotation to a given modelpart
+ * @details The class applies a rotation to a given modelpart given a constant
+ * angular velocity or it calculates the torque applied on the modelpart and
+ * calculates(by BDF2 time integration) the current theta and omega and rotates
```

Comment:
note that in MeshMoving also bossak is used, so maybe should be generalized in the future